scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,416 @@
+            -0.949107912342758524526189684047851,
+            -0.991455371120812639206854697526329
+        ]])
+
+        # 15-point weights
+        weights_15 = np.array([
+            0.022935322010529224963732008058970,
+            0.063092092629978553290700663189204,
+            0.104790010322250183839876322541518,
+            0.140653259715525918745189590510238,
+            0.169004726639267902826583426598550,
+            0.190350578064785409913256402421014,
+            0.204432940075298892414161999234649,
+            0.209482141084727828012999174891714,
+            0.204432940075298892414161999234649,
+            0.190350578064785409913256402421014,
+            0.169004726639267902826583426598550,
+            0.140653259715525918745189590510238,
+            0.104790010322250183839876322541518,
+            0.063092092629978553290700663189204,
+            0.022935322010529224963732008058970
+        ])
+
+        # 7-point nodes come from taking the 15-point nodes at the positions:
+        #   1, 3, 5, 7, 9, 11, 13
+        # (indexed from 0). The 7-point weights are
+        weights_7 = np.array([
+            0.129484966168869693270611432679082,
+            0.279705391489276667901467771423780,
+            0.381830050505118944950369775488975,
+            0.417959183673469387755102040816327,
+            0.381830050505118944950369775488975,
+            0.279705391489276667901467771423780,
+            0.129484966168869693270611432679082
+        ])
+
+        self.nodes = nodes_15
+        self.weights = weights_15
+
+        self.error_nodes = nodes_15
+        self.error_weights = np.array([
+            weights_15[0],
+            weights_15[1] - weights_7[0],
+            weights_15[2],
+            weights_15[3] - weights_7[1],
+            weights_15[4],
+            weights_15[5] - weights_7[2],
+            weights_15[6],
+            weights_15[7] - weights_7[3],
+            weights_15[8],
+            weights_15[9] - weights_7[4],
+            weights_15[10],
+            weights_15[11] - weights_7[5],
+            weights_15[12],
+            weights_15[13] - weights_7[6],
+            weights_15[14],
+        ])
+
+
+class Trapezoid(CubatureRuleLinearError):
+    def __init__(self):
+        self.nodes = np.array([
+            [-1],
+            [0],
+            [1]
+        ])
+        self.weights = np.array([
+            0.5,
+            1,
+            0.5
+        ])
+
+        # These come from the difference using 2 trapezoidal subregions versus three
+        # 1 trapezoidal subregions as an estimate for the error.
+        self.error_nodes = np.array([
+            [-1],
+            [0],
+            [1]
+        ])
+        self.error_weights = 1/3 * np.array([
+            -0.5,
+            1,
+            -0.5,
+        ])
+
+
+# TODO: is this a sensible default for maxiter?
+def cub(f, a, b, rule, rtol=1e-05, atol=1e-08, maxiter=10_000, args=()):
+    # TODO: This is how
```

Comment:
I think we should just pass args all the way through, so each `Rule` gains an `args` parameter. Just my thoughts.